File: solutions/45/45.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
         int n = nums.size();
        if (n <= 1) {
            return 0;
        }
    
        int jumps = 0; 
        int currentPos = 0; 
        int maxReach = 0;
    
        for (int i = 0; i < n - 1; i++) {
            maxReach = max(maxReach, i + nums[i]);

            if (i == currentPos) {
                jumps++;
                currentPos = maxReach;
            }
        }
    
        return jumps;
    }
};

File: solutions/45/45.swift
class Solution {
    func jump(_ nums: [Int]) -> Int {
        let n = nums.count
        if n <= 1 {
            return 0
        }
        
        var jumps = 0
        var currentPos = 0
        var maxReach = 0
        
        for i in 0..<(n - 1) {
            maxReach = max(maxReach, i + nums[i])
            if i == currentPos {
                jumps += 1
                currentPos = maxReach
            }
        }
        
        return jumps
    }
}